lldb/tools/lldb-dap/JSONUtils.cpp

```diff
@@ -1390,7 +1414,10 @@ llvm::json::Value CreateVariable(lldb::SBValue v, int64_t var_ref,
     object.try_emplace("variablesReference", 0);
 
   if (v.GetDeclaration().IsValid())
-    object.try_emplace("declarationLocationReference", var_ref);
+    object.try_emplace("declarationLocationReference", var_ref << 1);
+
+  if (HasValueLocation(v))
+    object.try_emplace("valueLocationReference", (var_ref << 1) | 1);
```

Comment:
Please create helper functions that do these bitwise manipulations. That would make it more obvious what's going on. E.g. varRefToDeclLocationRef and varRefToValueLocationRef